.github/workflows/stable-snapshot-timestamp.yml

```
@@ -30,10 +30,47 @@ on:
     paths:
       - 'repository/staged/root.json'
   workflow_dispatch:
+    inputs:
+      dry_run:
+        type: boolean
+        default: false
+        description: Does not trigger job, but checks on whether the job should run.
 
 jobs:
+  check:
+    # This job checks whether snapshot/timestamp should run.
+    runs-on: ubuntu-latest
+    outputs:
+      ceremony_wip: ${{ steps.check.outputs.wip }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Determine whether to run a snapshot/timestamp
+        id: check
+        run: |
+          BRANCHES=$(git branch  | grep ceremony/)
+          echo "${BRANCHES}"
+          # Check whether a ceremony was initiated within a week of the current date.
+          echo "wip=false" >> $GITHUB_OUTPUT
+          ceremony-re="^ceremony/[0-9]{4}-[0-9]{2}-[0-9]{2}$"
+          for branch in "${BRANCHES}"
+          do
+            if [[ "$branch" ~= "$ceremony-re" ]]; then
+              echo "found ceremony branch $branch"
+              branch_date=$(echo "$branch" | cut -d '/' -f2)
+              days_diff=$(( (`date -d $branch_date +%s` - `date -d "00:00" +%s`) / (24*3600) ))
+              if [[ "$days_diff" -lt 7 ]]; then
+                # Detected ceremony within 7 days of current date
+                echo "detected ceremony branch $branch within 7 days, stopping automated cron"
+                echo "wip=true" >> $GITHUB_OUTPUT
+              fi
+            fi
+          done
+
   run_snapshot_timestamp_publish:
-    if: (github.event_name == 'schedule' && github.repository == 'sigstore/root-signing') || (github.event_name != 'schedule')  # Don't run workflow in forks on cron
+    needs: check
+    if: (github.event_name == 'schedule' && github.repository == 'sigstore/root-signing' && needs.check.outputs.ceremony_wip == 'true') || (github.event_name != 'schedule' && inputs.dry_run == 'false')  # Don't run workflow in forks on cron
```

Catching my own mistake here!!

```suggestion
    if: (github.event_name == 'schedule' && github.repository == 'sigstore/root-signing' && needs.check.outputs.ceremony_wip == 'false') || (github.event_name != 'schedule' && inputs.dry_run == 'false')  # Don't run workflow in forks on cron
```